design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,319 @@
+
+A **DataSource** object references an import/clone source for a PVC. Functionally, it is similar to the 'source' field of a DataVolume. The DataSource’s spec is mutable, which has the side effect of allowing the DataSource to act as a link that will redirect to whatever source the link currently points to. In the future, the DataSource will act as a PVC Populator as well as a DataVolume source.
+
+**DataImportCron can manage a corresponding DataSource** to always point to the latest most up-to-date PVC that the cron has created. The DataImportCron should also be capable of adopting and taking ownership of an existing DataSource in order to statisfy the condition where an admin wishs to opt-in to using auto delivery/updates later on.
+
+
+## CDI: DataVolume DataSource Source Field
+
+The DataVolume object will add a new field capable of importing pvc content using a DataSource. 
+
+![DataVolume DataSource Field](golden-image-delivery-images/volume-reference-source.png "DataVolume DataSource Field")
+
+## SSP: Setup DataImportCrons for common templates
+
+Since the SSP operator manages installing common templates, this operator should also be responsible for setting up the DataImportCrons used to automatically source the disk images into the cluster that the common templates use.
+
+The location the DataImportCrons pull from will be configurable on the SSP operators Custom Resource.
+
+![SSP Disk Import Pipelines](golden-image-delivery-images/ssp-disk-import-pipelines.png "SSP Disk Import Pipelines")
+
+## Common templates: DataSources as DataVolume Source
+
+The common templates shipped with SSP need to have their DataVolumeTemplate sections transition to using a DataSource rather than the inlined PVC source field. This allows the DataImportCron to seamlessly update the underlying PVC the corresponding DataSource points to, which then ensures that the common templates are always using the most up-to-date disk images.
+
+## HCO: Integration with SSP Operator
+
+During installation, the HCO bundle's CSV will pass into the HCO operator (as ENV VARs) locations where disk images are sourced from for the automated disk image delivery/update feature. The end user can opt-in to automatic delivery/updates of these image sources through a tuning on the HCO's CR. From there the HCO is responsible for configuring DataImportCrons on the SSP operator's CR.
+
+This document doesn't include within it's scope how the HCO should expose this feature to the end user. The general expectation is that we will start with a simple tuning that enables toggling automated delivery/updates on and then consider advanced tunings in future releases. The exact structure of where this tuning belongs on the HCO API is left up to the HCO community.
```

Comment:
```suggestion
This document doesn't include within its scope how the HCO should expose this feature to the end user. The general expectation is that we will start with a simple tuning that disables automated delivery/updates and then consider advanced tunings in future releases. The exact structure of where this tuning belongs on the HCO API is left up to the HCO community.
```
Can we put the focus of this as on-by-default? I can see why most users would not like this machinery running, but I think that most users would prefer to have the most up-to-date base image on their cluster.

Reply:
@dankenigsberg during an update, turning disk delivery on by default will clobber usage of existing disks someone may have uploaded. That's why this is an opt-in feature. 

Think of the scenario where someone has strategically picked/uploaded the centos8 image they expect everyone to use as a base image, they update the HCO, then the centos8 base image all their templates use changes without them having any knowledge of this.

Reply:
the PR has now been merged, we can still continue this discussion point though if you'd like.